src/TmsIntegrationExample.Services/TmsGatewayApi/TmsGatewayClient.cs

```diff
@@ -0,0 +1,96 @@
+
+    public class TmsGatewayClient : ITmsGatewayClient
+    {
+        private readonly JsonSerializerOptions jsonSerializerOptions;
+        private readonly HttpClient httpClient;
+        private readonly TmsGatewayConfiguration tmsGatewayConfiguration;
+        private TokenResponse accessToken;
+
+        public TmsGatewayClient(
+            HttpClient httpClient,
+            IOptions<TmsGatewayConfiguration> tmsGatewayConfiguration,
+            JsonSerializerOptions jsonSerializerOptions)
+        {
+            this.tmsGatewayConfiguration = tmsGatewayConfiguration.Value;
+            this.httpClient = httpClient;
+            this.httpClient.BaseAddress = this.tmsGatewayConfiguration.BaseAddress;
+            this.jsonSerializerOptions = jsonSerializerOptions;
+        }
+
+        public async Task<Shipment> PostShipmentAsync(ShipmentPost shipment)
+        {
+            accessToken = await GetAccessTokenAsync();
+            httpClient.DefaultRequestHeaders.Clear();
+            httpClient.DefaultRequestHeaders.Add("Authorization", "Bearer " + accessToken.AccessToken);
+            httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+            var jsonContent = JsonConvert.SerializeObject(shipment);
+            var contentString = new StringContent(jsonContent, Encoding.UTF8, "application/json");
+            HttpResponseMessage response = await httpClient.PostAsync($"{tmsGatewayConfiguration.ShipmentsSubPath}", contentString);
+            response.EnsureSuccessStatusCode();
+            return await response.Content.ReadFromJsonAsync<Shipment>(jsonSerializerOptions);
+        }
+
+        public async Task<TokenResponse> GetAccessTokenAsync()
+        {
+            // Create an http client for all of the requests
+            var loginHttpClient = new HttpClient(new HttpClientHandler { AllowAutoRedirect = false }) { BaseAddress = this.tmsGatewayConfiguration.LoginBaseAddress };
+
+            // login
+            await loginHttpClient.PostAsync(
+                this.tmsGatewayConfiguration.LoginSubPath,
+                new StringContent(System.Text.Json.JsonSerializer.Serialize(new { this.tmsGatewayConfiguration.Email, this.tmsGatewayConfiguration.Password }), Encoding.UTF8, "application/json"));
+
+            // instantiate state, code verifier, and verifier challenge
+            var state = Helpers.CreateRandomString(128);
+            var verifier = Helpers.CreateRandomString(128);
+            var challenge = verifier.ToCodeChallenge();
+
+            // Create a relative url for requesting the authorization code
+            var url = new RequestUrl(this.tmsGatewayConfiguration.AuthorizeSubPath).CreateAuthorizeUrl(
+                responseType: "code",
+                clientId: this.tmsGatewayConfiguration.ClientId,
+                redirectUri: this.tmsGatewayConfiguration.RedirectUri,
+                scope: "openid profile engTech",
```

Comment:
Other than openid, scopes should be 100% configuration and, at this time, the engTech scope should be reserved specifically for ui client logins. TmsGateway shouldn't even permit it's use.

Reply:
added scope on appsettings file, can you please review?